provider/execution/execution.go

```diff
@@ -0,0 +1,371 @@
+				resolverErr = FromError(fmt.Errorf("table resolver panic: %s", r), WithResource(e.ResourceName), WithSeverity(diag.PANIC),
+					WithType(diag.RESOLVING), WithSummary("panic on resource table %s fetch", e.Table.Name))
+			}
+			close(res)
+		}()
+		if err := e.Table.Resolver(ctx, client, parent, res); err != nil {
+			if e.Table.IgnoreError != nil && e.Table.IgnoreError(err) {
+				client.Logger().Warn("ignored an error", "err", err, "table", e.Table.Name)
+				resolverErr = NewError(diag.IGNORE, diag.RESOLVING, e.ResourceName, "table[%s] resolver ignored error. Error: %s", e.Table.Name, err)
+			} else {
+				resolverErr = e.handleResolveError(client, err)
+			}
+		}
+	}()
+
+	nc := uint64(0)
+	for elem := range res {
+		objects := helpers.InterfaceSlice(elem)
+		if len(objects) == 0 {
+			continue
+		}
+		resolvedCount, dd := e.resolveResources(ctx, client, parent, objects)
+		// append any diags from resolve resources
+		diags = diags.Append(dd)
+		nc += resolvedCount
+	}
+	// check if channel iteration stopped because of resolver failure
+	if resolverErr != nil {
+		client.Logger().Error("received resolve resources error", "table", e.Table.Name, "error", resolverErr)
+		return 0, diags.Append(resolverErr)
+	}
+	// Print only parent resources
+	if parent == nil {
+		client.Logger().Info("fetched successfully", "table", e.Table.Name, "count", nc)
+	}
+	if err := e.cleanupStaleData(ctx, client, parent); err != nil {
+		return nc, diags.Append(FromError(err, WithType(diag.DATABASE), WithSummary("failed to cleanup stale data on table %s", e.Table.Name)))
+	}
+	return nc, diags
+}
+
+func (e *TableExecutor) resolveResources(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, objects []interface{}) (uint64, diag.Diagnostics) {
+	var (
+		resources = make(schema.Resources, 0, len(objects))
+		diags     diag.Diagnostics
+	)
+
+	for _, o := range objects {
+		resource := schema.NewResourceData(e.Db.Dialect(), e.Table, parent, o, e.extraFields, e.executionStart)
+		// Before inserting resolve all table column resolvers
+		if err := e.resolveResourceValues(ctx, meta, resource); err != nil {
+			e.Logger.Warn("skipping failed resolved resource", "reason", err.Error())
+			diags = diags.Append(err)
+			continue
+		}
+		resources = append(resources, resource)
+	}
+
+	// only top level tables should cascade
+	shouldCascade := parent == nil
+	resources, dbDiags := e.copyDataIntoDB(ctx, resources, shouldCascade)
+	diags = diags.Append(dbDiags)
+	totalCount := uint64(len(resources))
+
+	// Finally, resolve relations of each resource
+	for _, rel := range e.Table.Relations {
+		meta.Logger().Debug("resolving table relation", "table", e.Table.Name, "relation", rel.Name)
+		for _, r := range resources {
+			// ignore relation resource count
+			if _, innerDiags := e.WithTable(rel).Resolve(ctx, meta, r); innerDiags.HasDiags() {
+				diags = diags.Append(innerDiags)
```

Comment:
```suggestion
				diags = diags.Append(innerDiags...)
```

Reply:
the .Append function does this automatically

Reply:
oh... is `Append` a good name for it (ie. promising order) what about `.Add`?

Reply:
changed